widgetssdk/src/main/java/com/glia/widgets/callvisualizer/ActivityWatcherForDialogs.kt

```diff
@@ -0,0 +1,297 @@
+            positiveButtonClickListener = {
+                dismissAlertDialog()
+                dialogController.dismissCurrentDialog()
+                NotificationManager.openNotificationChannelScreen(contextWithStyle)
+            },
+            negativeButtonClickListener = {
+                dismissAlertDialog()
+                dialogController.dismissCurrentDialog()
+            },
+            cancelListener = {
+                it.dismiss()
+                dialogController.dismissCurrentDialog()
+            }
+        )
+    }
+
+    private fun showAllowScreenSharingNotificationsAndStartSharingDialog(resumedActivity: WeakReference<Activity?>) {
+        val activity = resumedActivity.get() ?: return
+        if (alertDialog != null && alertDialog!!.isShowing) {
+            return
+        }
+
+        Logger.d(TAG, "Show screen sharing and notifications dialog")
+        val builder = UiTheme.UiThemeBuilder()
+        val theme = builder.build()
+        val contextWithStyle = prepareContextWithStyle(activity)
+        alertDialog = Dialogs.showOptionsDialog(
+            context = contextWithStyle,
+            theme = theme,
+            title = activity.getString(R.string.glia_dialog_screen_sharing_offer_enable_notifications_title),
+            message = activity.getString(R.string.glia_dialog_screen_sharing_offer_enable_notifications_message),
+            positiveButtonText = activity.getString(R.string.glia_dialog_screen_sharing_offer_enable_notifications_yes),
+            negativeButtonText = activity.getString(R.string.glia_dialog_screen_sharing_offer_enable_notifications_no),
+            positiveButtonClickListener = {
+                dismissAlertDialog()
+                dialogController.dismissCurrentDialog()
+                NotificationManager.openNotificationChannelScreen(contextWithStyle)
+            },
+            negativeButtonClickListener = {
+                dismissAlertDialog()
+                dialogController.dismissCurrentDialog()
+                screenSharingController.onScreenSharingDeclined()
+            },
+            cancelListener = {
+                it.dismiss()
+                dialogController.dismissCurrentDialog()
+                screenSharingController.onScreenSharingDeclined()
+            }
+        )
+    }
+
+    private fun showScreenSharingDialog(resumedActivity: WeakReference<Activity?>) {
+        val activity = resumedActivity.get() ?: return
+        if (alertDialog != null && alertDialog!!.isShowing) {
+            return
+        }
+
+        Logger.d(TAG, "Show screen sharing dialog")
+        val builder = UiTheme.UiThemeBuilder()
+        val theme = builder.build()
+        val contextWithStyle = prepareContextWithStyle(activity)
+
+        alertDialog = Dialogs.showScreenSharingDialog(
+            contextWithStyle,
+            theme,
+            activity.getText(R.string.glia_dialog_screen_sharing_offer_title).toString(),
```

Comment:
why not use `activity.getString()` instead of `activity.gettext(...).toString()`

Reply:
Yes, for sure. Copied part of the code from another class and haven't noticed that it should be changed.

Reply:
Fixed 909b913c3b5428cff06a6414b554d104d1bcf1e5